lib/widget/lottie/web/lottiestate.dart

```diff
@@ -14,115 +18,201 @@ import 'dart:js' as js;
+      AnimationStatus.forward: controller.onForward,
+      AnimationStatus.reverse: controller.onReverse,
+      AnimationStatus.dismissed: controller.onStop,
+      AnimationStatus.completed: controller.onComplete,
+    };
+
+    controller.lottieController!.addStatusListener(
+      (status) {
+        if (animationStatusActionMap[status] != null) {
+          ScreenController().executeAction(
+            context,
+            animationStatusActionMap[status]!,
+            event: EnsembleEvent(widget),
+          );
+        }
+      },
+    );
+  }
+
   @override
   Widget buildWidget(BuildContext context) {
     final isCanvasKit = js.context['flutterCanvasKit'] != null;
-    print('isCanvasKit=$isCanvasKit');
-    BoxFit? fit = WidgetUtils.getBoxFit(widget.controller.fit);
+    final controller = widget.controller;
 
+    BoxFit? fit = WidgetUtils.getBoxFit(controller.fit);
     Widget rtn = BoxWrapper(
-        widget: isCanvasKit ? buildLottieCanvas(fit) : buildLottieHtml(fit),
-        boxController: widget.controller,
-        ignoresMargin: true,
-        ignoresDimension: true);
-    if (widget.controller.onTap != null) {
+      widget: isCanvasKit
+          ? buildLottieCanvas(fit, controller)
+          : buildLottieHtml(fit, controller),
+      boxController: controller,
+      ignoresMargin: true,
+      ignoresDimension: true,
+    );
+
+    if (controller.onTap != null) {
       rtn = GestureDetector(
-          child: rtn,
-          onTap: () => ScreenController().executeAction(
-              context, widget.controller.onTap!,
-              event: EnsembleEvent(widget)));
+        child: rtn,
+        onTap: () => ScreenController().executeAction(
+          context,
+          controller.onTap!,
+          event: EnsembleEvent(
+            widget,
+          ),
+        ),
+      );
     }
-    if (widget.controller.margin != null) {
-      rtn = Padding(padding: widget.controller.margin!, child: rtn);
+
+    if (controller.margin != null) {
+      rtn = Padding(padding: controller.margin!, child: rtn);
     }
+
     return rtn;
   }
 
-  Widget buildLottieHtml(BoxFit? fit) {
-    String source = widget.controller.source.trim();
-    double width = widget.controller.width?.toDouble() ?? 250;
-    double height = widget.controller.height?.toDouble() ?? 250;
-    String _id = widget.controller.id ?? id;
-    String repeat = (widget.controller.repeat == null)
-        ? 'true'
-        : widget.controller.repeat!.toString();
-    String tag = '<div id="$_id"></div>';
-    if (widget.controller.width == null || widget.controller.height == null) {
-      print(
-          'Lottie widget must have explicit width and height for html renderer in the browser (this includes the preview in the editor). You do not need to specify width or height for native apps or for canvaskit renderer. Defaulting to $width for width and $height for height');
+  Widget buildLottieHtml(BoxFit? fit, LottieController controller) {
```

Comment:
widget.controller is available everywhere inside the state. Why then create a variable called controller and then pass it around. 

Most of these changes are unnecessary and make it very hard to actually review the change. Please focus only on the actual change. Remove these unnecessary changes and update your PR